actors/miner/tests/prove_commit_niporep.rs

```diff
@@ -0,0 +1,159 @@
+use fvm_ipld_encoding::ipld_block::IpldBlock;
+use fvm_shared::{bigint::BigInt, clock::ChainEpoch, error::ExitCode};
+
+use fil_actor_miner::{Actor, Method, SectorOnChainInfo, SectorOnChainInfoFlags};
+use num_traits::Zero;
+use util::*;
+
+mod util;
+
+const PERIOD_OFFSET: ChainEpoch = 100;
+
+#[test]
+fn prove_zero_sectors_ni_fail() {
+    let h = ActorHarness::new(PERIOD_OFFSET);
+    let rt = h.new_runtime();
+    rt.balance.replace(BIG_BALANCE.clone());
+    let miner = rt.receiver.id().unwrap();
+
+    let seal_randomness_epoch = PERIOD_OFFSET + 1;
+    let expiration = seal_randomness_epoch + 1000;
+
+    rt.set_epoch(seal_randomness_epoch);
+    h.construct_and_verify(&rt);
+
+    rt.set_epoch(seal_randomness_epoch + 400);
+
+    let params = h.make_prove_commit_ni_params(miner, &[], seal_randomness_epoch, expiration, 0);
+
+    let res = rt.call::<Actor>(
+        Method::ProveCommitSectorsNI as u64,
+        IpldBlock::serialize_cbor(&params).unwrap(),
+    );
+
+    assert!(res.is_err());
+    assert_eq!(res.unwrap_err().exit_code(), ExitCode::USR_ILLEGAL_ARGUMENT);
+}
+
+#[test]
+fn prove_one_sector_aggregate_ni() {
+    let h = ActorHarness::new(PERIOD_OFFSET);
+    let rt = h.new_runtime();
+    rt.balance.replace(BIG_BALANCE.clone());
+    let miner = rt.receiver.id().unwrap();
+
+    let seal_randomness_epoch = PERIOD_OFFSET + 1;
+    let expiration = seal_randomness_epoch + 1000;
+
+    rt.set_epoch(seal_randomness_epoch);
+    h.construct_and_verify(&rt);
+
+    rt.set_epoch(seal_randomness_epoch + 400);
+
+    let sector_nums = (0..1).collect::<Vec<_>>();
+    let params =
+        h.make_prove_commit_ni_params(miner, &sector_nums, seal_randomness_epoch, expiration, 0);
+
+    let res = h.prove_commit_sectors_ni(&rt, params, true);
+
+    assert!(res.is_ok());
+
+    let deadlines = h.get_state(&rt).load_deadlines(&rt.store).unwrap();
+    let deadline = deadlines.load_deadline(&rt.store, 0).unwrap();
+    assert_eq!(deadline.live_sectors, 1);
+}
+
+#[test]
+fn prove_sectors_max_aggregate_ni() {
+    let h = ActorHarness::new(PERIOD_OFFSET);
+    let rt = h.new_runtime();
+    rt.balance.replace(BIG_BALANCE.clone());
+    let miner = rt.receiver.id().unwrap();
+
+    let seal_randomness_epoch = PERIOD_OFFSET + 1;
+    let expiration = seal_randomness_epoch + 1000;
+    let proving_deadline = 42;
+
+    rt.set_epoch(seal_randomness_epoch);
+    h.construct_and_verify(&rt);
+
+    let activation_epoch = seal_randomness_epoch + 400;
+    rt.set_epoch(activation_epoch);
+
+    for i in 0..3 {
```

Comment:
Why does this test iterate three times? Could you please leave a comment explaining?

Reply:
Added separate test that check if multiple commits are all adding sectors to the same deadline.